File: DHealth-Android/Java/app/src/main/java/com/example/hamid/dhealth/di/module/ActivityModule.java
package com.example.hamid.dhealth.di.module;

import com.example.hamid.dhealth.ActiveLedgerHelper;
import com.example.hamid.dhealth.ui.Activities.DashboardScreen;
import com.example.hamid.dhealth.ui.Activities.DoctorPatientDescriptionActivity;
import com.example.hamid.dhealth.ui.Activities.LoginScreen;
import com.example.hamid.dhealth.ui.Activities.ProfileScreen;
import com.example.hamid.dhealth.ui.Activities.ReportDetailActivity;
import com.example.hamid.dhealth.ui.Activities.SignUpScreen;
import com.example.hamid.dhealth.ui.Activities.SplashActivity;
import com.example.hamid.dhealth.ui.Activities.UploadFileActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityModule {

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract DashboardScreen contributeDashboardActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract LoginScreen contributeLoginActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract SignUpScreen contributeSignUpActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract ProfileScreen contributeProfileActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract ReportDetailActivity contributeReportDetailActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract UploadFileActivity contributeUploadFileActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract SplashActivity contributeSplashActivity();

    @ContributesAndroidInjector(modules = FragmentModule.class)
    abstract DoctorPatientDescriptionActivity contributeDoctorPatientDescriptionActivity();

//    @ContributesAndroidInjector(modules = FragmentModule.class)
//    abstract ActiveLedgerHelper contributeActiveLedgerHelper();
}

File: DHealth-Android/Kotlin/app/src/main/java/com/example/hamid/dhealth/di/module/ActivityModule.kt
package com.example.hamid.dhealth.di.module

import com.example.hamid.dhealth.ui.Activities.*
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityModule {

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeDashboardActivity(): DashboardScreen

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeLoginActivity(): LoginScreen

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeSignUpActivity(): SignUpScreen

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeProfileActivity(): ProfileScreen

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeReportDetailActivity(): ReportDetailActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeUploadFileActivity(): UploadFileActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeSplashActivity(): SplashActivity

    @ContributesAndroidInjector(modules = [FragmentModule::class])
    internal abstract fun contributeDoctorPatientDescriptionActivity(): DoctorPatientDescriptionActivity

}
